File: 1105a/main.py
import sys

input = lambda: sys.stdin.readline().rstrip()


def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

a = II()
L = LII()
x = 0
ans = 1 << 60
for i in range(1, 101):
    tmp = 0
    for j in L:
        if abs(j - i) > 1:
            tmp += abs(j - i) - 1

    if tmp < ans:
        ans = tmp
        x = i
print(x, ans)

File: 1105a/main.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"math"
	"os"
)

func solve(in *bufio.Reader, out *bufio.Writer) {
	var b int
	Fscan(in, &b)
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
func run(r io.Reader, w io.Writer) {
	in := bufio.NewReader(r)
	out := bufio.NewWriter(w)
	defer out.Flush()
	var a int
	Fscan(in, &a)
	L := make([]int, a)
	for i := 0; i < a; i++ {
		Fscan(in, &L[i])
	}

	ans := math.MaxInt
	x := 0
	for i := 1; i < 101; i++ {
		tmp := 0
		for _, j := range L {
			if abs(j-i) > 1 {
				tmp += abs(j-i) - 1
			}
		}
		if tmp < ans {
			ans = tmp
			x = i
		}
	}
	Fprint(out, x, ans)

}

func main() {
	run(os.Stdin, os.Stdout)
}
